### PR TITLE
Support binlog_row_image=MINIMAL

### PR DESCRIPTION
**Pull request by osheroff**
The bulk of the work in this changes was in creating a unified iterator that can 
traverse Columns, ColumnDefinitions, and the "which columns are actually present" bitmap in 
one go.

@zendesk/rules

### CODE REVIEW

File: docs/docs/compat.md [hunk 1]
```diff
@@ -9,7 +9,17 @@
 ### Unsupported
 
 - Mysql 5.7 is untested with Maxwell, and in particular GTID replication is unsupported as of yet.
-- `binlog_row_image=MINIMAL` is not supported and will break Maxwell in a variety of amusing ways.
+
+### binlog_row_image=MINIMAL
```
Comment (wushujames) [inline docs/docs/compat.md]: How does binlog_row_image=MINIMAL work with log compaction? Or, does it not work with log compaction? I guess log compaction isn't technically a Maxwell concern, it's a Kafka concern, so it's up to the user if they care about it.

File: src/test/resources/sql/json/test_minimal [hunk 2]
```diff
@@ -1,36 +1,24 @@
...
+insert into test_table set a='avalue2', b='bvalue2'
 
-->  {"database": "test_db_1", "table": "test_table", "type": "insert", "data": { "id": 2, "textcol": "test_col_2"} }
+->  {"database": "test", "table": "test_table", "type": "insert", "data": {"id": 2, "a": "avalue2", "b": "bvalue2"} }
 
-alter table test_table add column `datecol` datetime NULL default null AFTER id
+update test_table set a='avalue3' where id = 2
 
-insert into test_table set textcol='test_col_2', datecol='1979-10-01 00:00:00'
+->  { "database": "test", "table": "test_table", "type": "update", "data": {"id": 2, "a": "avalue3"} }
```
Comment (vanchi-zendesk) [inline src/test/resources/sql/json/test_minimal]: suggested test:
```
update test_table set a='avalue3' where a='avalue2'
```

Comment (vanchi-zendesk) [inline src/test/resources/sql/json/test_minimal]: nm.

**Comment by vanchi-zendesk** [comment]
:+1:

**Pull request merged**